.travis.yml

```diff
@@ -1,18 +1,21 @@
-sudo: required 
-dist: trusty 
+sudo: required
+dist: trusty
 language: generic
 notifications:
   email:
     on_success: change
     on_failure: always
 env:
   global:
-    - ROS_DISTRO="indigo"
     - UPSTREAM_WORKSPACE=file
-    - ROSINSTALL_FILENAME=.travis.rosinstall
   matrix:
-    - ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
+    - ROS_DISTRO="indigo" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
+    - ROS_DISTRO="jade" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
+matrix:
+  allow_failures:
+    - env: ROS_DISTRO="jade" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
```

Comment:
the idea is the following:
 - try to use `indigo_dev` for jade and kinetic
 - if it works for kinetic, fine!
 - if it fails, then create a `kinetic_dev` branch and make it work for kinetic (in this case we do not care anymore about jade - because we do not support it)

so tests are only forward...not backwards...